src/gui/main_gui.cpp

```diff
@@ -145,9 +147,9 @@ GUITest(World world) {
 
     std::vector<glm::ivec3> model_matrices;
     // generate positions of trees
-    for (unsigned int x = 0; x < world.terrain_main.get_X_MAX(); x += 40)
-        for (unsigned int y = 0; y < world.terrain_main.get_Y_MAX(); y += 40) {
-            unsigned int z = world.terrain_main.get_Z_solid(x, y) + 1;
+    for (uint32_t x = 0; x < world.terrain_main.get_X_MAX(); x += 40)
+        for (uint32_t y = 0; y < world.terrain_main.get_Y_MAX(); y += 40) {
+            uint32_t z = world.terrain_main.get_Z_solid(x, y) + 1;
```

Comment:
USE A `size_t`

All this does is use `%eax` instead of `%rax`. Those upper bits are just sitting unused. You're not saving any space by using uint32s, they **will** be assigned to registers (as they are hot variables) and you'll just leave half the reg unused